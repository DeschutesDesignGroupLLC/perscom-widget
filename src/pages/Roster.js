import * as Sentry from '@sentry/react'
import React from 'react'
import useQuery from '../api/APIUtils'
import { Loading } from '../components/Loading'
import { Table } from '../components/Table'
import { config } from '../constants'
import { Link } from '../components/Link'
import { Alert } from '../components/Alert'

function Roster() {
  const { data, loading, error } = useQuery({
    url: config.roster.API_URL
  })

  return (
    <>
      {loading ? (
        <Loading />
      ) : (
        <>
          {error ? (
            <Alert message={error} type='danger' />
          ) : data && !!data.length ? (
            <div className='flex flex-col space-y-4'>{data.map((unit) => renderUnit(unit))}</div>
          ) : (
            <Alert message='No units found. Please add a unit to view the roster.' />
          )}
        </>
      )}
    </>
  )
}

function renderUnit(unit) {
  const { id, name, users } = unit

  return (
    <Table
      key={id}
      columns={[
        {
          name: name,
          headerAttributes: { colSpan: '6' },
          cellClasses: ['w-1/6', 'whitespace-normal', 'sm:whitespace-nowrap', '!py-3'],
          cellContent: (user) => {
            const { name, rank, position, id: user_id } = user
            const { image, abbreviation, name: rank_name } = rank ?? {}
            const { image_url } = image ?? {}
            const { name: position_name } = position ?? {}
            return (
              <div className='flex items-center'>
                {rank && (
                  <div className='flex items-center w-6 sm:w-8 flex-shrink-0'>
                    {image_url ? (
                      <img className='w-6 sm:w-8 font-bold' src={image_url} alt={rank_name} />
                    ) : (
                      <div className='font-bold text-sm'>{abbreviation}</div>
                    )}
                  </div>
                )}
                <div className='flex flex-col ml-4'>
                  <Link
                    href={`/users/${user_id}`}
                    className='text-sm font-medium text-gray-900 dark:text-white hover:text-gray-500 active:text-blue-600'
                  >
                    {name}
                  </Link>
                  <div className='md:hidden text-xs text-gray-500 dark:text-gray-400'>{position_name}</div>
                </div>
              </div>
            )
          }
        },
        {
          name: 'Position',
          key: 'position.name',
          hidden: true,
          cellClasses: ['hidden', 'md:table-cell', 'w-1/6', 'whitespace-normal', 'lg:whitespace-nowrap', '!py-3']
        },
        {
          name: 'Specialty',
          hidden: true,
          key: 'specialty.name',
          cellClasses: ['hidden', 'sm:table-cell', 'w-1/6', 'whitespace-normal', 'lg:whitespace-nowrap', '!py-3']
        },
        {
          name: 'Status',
          hidden: true,
          cellClasses: ['text-right', 'md:text-center', 'whitespace-nowrap', 'w-1/12', '!py-3'],
          cellContent: (user) => {
            const { status } = user
            return (
              <>
                {status && (
                  <span className={`inline-flex rounded-full px-2 text-xs font-semibold leading-5 ${status.color}`}>{status.name}</span>
                )}
              </>
            )
          }
        },
        {
          name: 'Online',
          hidden: true,
          cellClasses: ['hidden', 'lg:table-cell', 'text-center', 'w-1/12', '!py-3'],
          cellContent: (user) => {
            const { online } = user
            return (
              <>
                {online ? (
                  <span className='inline-flex rounded-full px-2 text-xs font-semibold leading-5 bg-green-100 text-green-600'>Online</span>
                ) : (
                  <span className='inline-flex rounded-full px-2 text-xs font-semibold leading-5 bg-sky-100 text-sky-600'>Offline</span>
                )}
              </>
            )
          }
        },
        {
          name: 'Link',
          hidden: true,
          cellClasses: ['hidden', 'xl:table-cell', 'text-center', 'w-1/6', '!py-3'],
          cellContent: (user) => {
            const { name, id } = user
            return (
              <Link href={`/users/${id}`} className='text-gray-500 hover:text-gray-700 active:text-blue-600'>
                Personnel Profile<span className='sr-only'>, {name}</span>
              </Link>
            )
          }
        }
      ]}
      rows={users}
    />
  )
}

export default Sentry.withProfiler(Roster)
