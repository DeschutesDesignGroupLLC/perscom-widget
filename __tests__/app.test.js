/**
 * @jest-environment jsdom
 */
import { expect, it, describe, beforeEach, beforeAll, jest } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import Layout from '../src/app/layout';
import Newsfeed from '../src/app/newsfeed/page';
import Roster from '../src/app/roster/page';

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      prefetch: () => jest.fn()
    };
  },
  usePathname() {
    return 'test.com';
  }
}));

const searchParams = {
  apikey: 'foo',
  perscomid: 'bar'
};

beforeAll(() => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn()
    }))
  });
});

describe('Roster', () => {
  beforeEach(() => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        status: 200,
        json: () =>
          Promise.resolve({
            data: [
              {
                id: 1,
                name: 'group foobar',
                units: [
                  {
                    id: 1,
                    name: 'unit foobar',
                    users: [
                      {
                        id: 1,
                        name: 'user foobar'
                      }
                    ]
                  }
                ]
              }
            ]
          })
      })
    );
  });

  it('can render', async () => {
    const { container } = render(
      Layout({
        children: await Roster(searchParams)
      })
    );

    expect(screen.getByTestId('unit foobar')).toBeInTheDocument();
    expect(screen.getByTestId('user foobar')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});

describe('Newsfeed', () => {
  beforeEach(() => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        status: 200,
        json: () =>
          Promise.resolve({
            data: [
              {
                id: 1,
                headline: 'newsfeed foobar'
              }
            ]
          })
      })
    );
  });

  it('can render', async () => {
    const { container } = render(
      Layout({
        children: await Newsfeed(searchParams)
      })
    );

    expect(screen.getByTestId('newsfeed foobar')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
