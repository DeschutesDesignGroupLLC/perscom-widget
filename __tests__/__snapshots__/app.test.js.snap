// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsfeed can render 1`] = `
<div>
  <html
    lang="en"
  >
    <body
      class="text-gray-500 dark:text-gray-400 className"
      style="margin: 0.2rem;"
    >
      <script>
        !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
      </script>
      <main>
        <div
          class="flex flex-col space-y-2"
        >
          <div
            class="flex flex-col space-y-6"
          >
            <div
              class="flex flex-col space-y-4"
              data-testid="newsfeed foobar"
            >
              <div
                class="flex flex-row items-center space-x-2 px-1 text-sm"
              >
                <div
                  class="flex justify-center items-center space-x-4 rounded"
                  data-testid="flowbite-avatar"
                >
                  <div
                    class="relative"
                  >
                    <div
                      class="rounded-full ring-2 ring-gray-300 dark:ring-gray-500 w-6 h-6 relative overflow-hidden bg-gray-100 dark:bg-gray-600"
                      data-testid="flowbite-avatar-img"
                    >
                      <svg
                        class="absolute w-auto h-auto text-gray-400 -bottom-1"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="text-sm font-normal"
                >
                  <span
                    class="font-bold"
                  >
                    User
                  </span>
                   
                  added a new message
                   â€¢
                   
                  <span
                    class="font-light"
                  />
                </div>
              </div>
              <div
                class="flex rounded-lg border border-gray-200 bg-white shadow-sm dark:border-gray-700 dark:bg-gray-800 flex-col !gap-2 p-6"
                data-testid="flowbite-card"
                target="_blank"
              >
                <div
                  class="flex h-full flex-col justify-center gap-4 p-0"
                >
                  <div
                    class="flex flex-col space-y-2"
                  >
                    <div
                      class="flex flex-col items-start justify-start space-y-2 sm:flex-row sm:items-center sm:justify-between sm:space-y-0"
                    >
                      <a
                        rel="noreferrer"
                        target="_blank"
                      >
                        <div
                          class="flex flex-row items-center space-x-2"
                        >
                          <h5
                            class="text-sm font-semibold uppercase sm:leading-6 cursor-default"
                          >
                            newsfeed foobar
                          </h5>
                        </div>
                      </a>
                      <div
                        class="hidden text-xs text-gray-400 dark:text-gray-500 md:block"
                      >
                        Invalid Date
                      </div>
                    </div>
                    <div>
                      <div
                        class="text-sm"
                      >
                        No content exists for this item.
                      </div>
                    </div>
                    <div
                      class="block text-xs text-gray-400 dark:text-gray-500 md:hidden"
                    >
                      Invalid Date
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
      <footer
        class="flex flex-row justify-center"
      >
        <div
          class="mt-2 flex items-center space-x-1"
        >
          <svg
            class="h-6 w-16"
            height="50"
            version="1.2"
            viewBox="0 0 215 50"
            width="215"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              PERSCOM
            </title>
            <g
              id="background-logo"
            >
              <path
                d="m-45-102.3h366.9v292.6h-366.9z"
                id="Layer"
                style="fill: none;"
              />
            </g>
            <g
              id="text-logo-path-group"
            >
              <g
                id="Layer"
              >
                <g
                  id="text-logo-path-0"
                >
                  <path
                    d="m81.9 39.3l-7.2 1.4q4.3-18.7 5.8-29.1 4.5-1.3 6.7-1.3 2.2 0 3.5 0.6 1.2 0.6 2 1.7 1.5 1.9 1.5 5.5 0 2.2-0.9 4.4-0.8 2.2-2.2 3.8-1.5 1.7-3.3 2.7-1.9 1-3.8 1h-0.5q-0.8 3.3-1.6 9.3zm2.5-13.7q1.4-0.3 2.7-2.3 1.3-2.2 1.3-4.9 0-1.5-0.7-2.3-0.6-0.8-1.4-0.8-1.1 5.8-1.9 10.3zm19.8 13.1l-13.4 2 5.4-29.2 13.1-1.2q-0.3 1-0.5 1.8-0.2 0.9-0.4 1.7-0.2 0.9-0.3 1.5-1.2 0.1-3.3 0.1h-2.8l-1.5 7.4q2.2-0.1 3.7-0.1 1.5-0.1 2-0.1-0.2 1.2-0.3 2.3-0.2 1.1-0.4 2.2-0.4 0.1-1.3 0.1-1 0-2 0.1-1 0-2.2 0l-1.4 7.6 6-0.9zm7.2 0.3l-6.7 1.7q4.2-20.2 5.1-28.7 4.7-1.4 10.5-1.4 4.4 0 5.2 3 0.2 0.9 0.2 2.3 0 1.3-0.5 2.9-0.6 1.6-1.6 3.1-2 2.9-4.9 4.5 1.6 3.4 2.8 6.3 1 2.4 1.7 4.2l2.9-5.8q0.6 1.1 2.2 2 1.6 1 2.4 1 0.9 0 1.4-0.2 0.5-0.1 1-0.4 1-0.8 1-1.8 0-1-0.7-1.9-0.6-0.8-1.5-1.6-1-0.7-2.1-1.5-1.1-0.8-2-1.7-2.3-2-2.3-4.2 0-2.2 1.3-4.1 1.2-1.8 3.2-3.2 1.9-1.3 4.2-2.1 2.2-0.8 3.8-0.8 1.6 0 2.6 0.3 1 0.3 1.7 0.8 1.6 1.3 1.6 3.8 0 0.6-0.1 1.2-0.1 0.5-0.2 1.2l-6 1.8q0.2-0.8 0.2-1.7 0-0.9-0.4-1.5-0.5-0.5-1-0.5-0.5 0-1 0.2-0.5 0.3-0.9 0.7-1.2 0.9-1.2 2 0 1 0.6 1.7 0.6 0.7 1.4 1.2 0.8 0.6 1.8 1.2 0.9 0.6 1.8 1.5 1.9 2.3 1.9 6 0 4.5-3.3 7.5-3.1 2.8-7.9 2.8-3.4 0-6.1-3.3 0 0.2 0.1 0.3l-6.4 2.2q-3.2-10.9-3.3-12.4l-0.5 0.2q-1.3 7.4-2 11.2zm4.4-22.7l-1.4 6.6q1.7-0.2 4.1-2.1 2.6-1.9 2.6-3.1 0-1.2-2.3-1.4-0.6 0-1.2 0zm27.5 22.1q-1.6-2.5-1.6-6.6 0-4 1-7.8 1.1-3.8 3-6.8 2-3 4.6-4.8 2.7-1.9 5.9-1.9 2.1 0 3.5 1.6 1.4 1.6 1.4 4.2v0.5q0 0.3 0 0.6l-5.4 1.4q-0.5-1-1.2-1-0.6 0-1.4 0.8-0.7 0.7-1.4 1.9-0.7 1.3-1.3 2.8-0.7 1.5-1.2 3-1.1 3.4-1.1 5.4 0 2.1 1.3 2.1 1 0 2.1-1.2 1.1-1.3 1.4-2.7l4.8 1.9q-1.9 4.2-4.8 6.7-2.6 2.2-5.4 2.2-2.5 0-4.2-2.3zm14.7-6q0-4.6 2.2-9.9 2.1-5.3 5.6-8.9 3.7-3.7 7.8-3.7 3.4 0 5 2.7 1.2 2.2 1.2 6.7 0 2.5-0.7 5.4-0.7 3-1.9 5.5-1.2 2.5-2.8 4.5-1.6 1.9-3.3 3.3-3.6 2.8-6.9 2.8-2.8 0-4.4-2-1.8-2.2-1.8-6.4zm6.2-1.1q0 1.6 1.3 2.4 0.4 0.3 1 0.3 1.8 0 3.2-1.3 1.5-1.2 2.4-3.1 1-1.8 1.5-3.9 0.5-2.1 0.5-3.9 0-3.3-1.9-3.3-2.5 0-4.5 2.4-1.5 1.9-2.7 5.3-0.8 2.7-0.8 5.1zm18.2 8.7l-6.8 0.4q6.6-19.8 8.9-28.9 2.3 0 4.3-0.3 2-0.2 4.4-0.4-0.2 0.9-0.3 2-0.1 1.1-0.3 2.4-0.2 1.2-0.4 2.5-0.2 1.3-0.5 3.4-0.3 2.1-0.8 5.2h0.6q4.6-12.5 6.1-15.6 2.1-0.2 3.4-0.4 1.3-0.3 2.2-0.4 0.9-0.1 2-0.3-2 11.6-3.3 29.4l-7.2 1.2q0-1.4 0.4-4.6l0.5-4.8q0.4-3.7 0.8-5.7h-0.4q-0.5 1.7-1.2 3.6-0.8 1.9-1.5 3.9-0.7 2-1.4 3.9-0.7 1.9-1.2 3.6-1.5 0.1-2.5 0.1l-2.4 0.2 1.3-15.3h-0.6q-0.1 0.5-0.4 1.8-0.4 1.2-0.8 2.9-0.4 1.6-0.9 3.3-0.5 1.8-1 3.3-0.7 2.6-1 3.6z"
                    fill-rule="evenodd"
                    id="Layer"
                    style="fill: #646464; stroke: #ffffff; stroke-width: 0;"
                  />
                </g>
              </g>
            </g>
            <g
              id="shape-4ba72ef2-936a-49d3-93b3-1ec8e54791af"
            >
              <g
                id="Layer"
              >
                <g
                  id="path-1"
                >
                  <path
                    d="m51.2 14.9l-3 17.9c-0.8 4.9-3.6 9.2-7.6 11.5l-4.3 2.5-4.3-2.5c-4-2.3-6.9-6.6-7.7-11.5l-3-17.9 15-5.8zm-4.3-7.1l-0.9-0.9 1.3-0.2 0.5-1.1 0.6 1.1 1.3 0.2-0.9 0.9 0.2 1.3-1.2-0.6-1.1 0.6zm-12-4.3l-1.4-1.3 1.9-0.2 0.8-1.7 0.8 1.7 1.8 0.2-1.3 1.3 0.3 1.8-1.6-0.8-1.7 0.8zm-11.2 4.3l-0.9-0.9 1.3-0.2 0.5-1.1 0.6 1.1 1.3 0.2-0.9 0.9 0.2 1.3-1.2-0.6-1.1 0.6z"
                    id="Layer"
                    style="fill: #d9d9d9;"
                  />
                </g>
                <g
                  id="path-2"
                >
                  <path
                    d="m52.3 22.9l-0.3 1.4 15.2-4.7-4.3 7.7-12 3.7-0.2 1.4 11.4-3.6-4.3 7.7-9.4 2.9c-1.6 2.9-3.9 5.3-6.7 6.9l-5.4 3.2-5.5-3.2c-2.7-1.6-5-4-6.6-6.9l-9.4-2.9-4.3-7.7 11.4 3.6-0.3-1.4-12-3.7-4.3-7.7 15.2 4.7-0.2-1.4-15.9-4.9-4.2-7.7 19 5.9-0.5-2.8 17.6-6.8 17.5 6.8-0.4 2.8 19-5.9-4.3 7.7zm-2.8 10.1l3.2-18.9-16.4-6.3-16.4 6.3 3.2 18.9c0.9 5.3 4 9.9 8.3 12.4l4.9 2.8 4.9-2.8c4.3-2.5 7.4-7.1 8.3-12.4z"
                    fill-rule="evenodd"
                    id="Layer"
                    style="fill: #d9d9d9;"
                  />
                </g>
              </g>
            </g>
          </svg>
          <div
            class="text-xs dark:text-gray-500"
          >
            Widget Provided By
             
            <a
              class="underline"
              href="https://perscom.io"
              rel="noreferrer"
              target="_blank"
            >
              PERSCOM.io
            </a>
            .
             
            <span
              class="hidden sm:inline"
            >
              Copyright 
              2024
               Deschutes Design Group LLC.
            </span>
             
            <span
              class="hidden md:inline"
            >
              Version 
              1.0.0
              .
            </span>
          </div>
        </div>
      </footer>
    </body>
  </html>
</div>
`;

exports[`Roster can render 1`] = `
<div>
  <html
    lang="en"
  >
    <body
      class="text-gray-500 dark:text-gray-400 className"
      style="margin: 0.2rem;"
    >
      <script>
        !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
      </script>
      <main>
        <div
          class="flex rounded-lg border border-gray-200 bg-white shadow-sm dark:border-gray-700 dark:bg-gray-800 flex-col"
          data-testid="flowbite-card"
        >
          <div
            class="flex h-full flex-col justify-center gap-4 p-0"
          >
            <div
              class="flex flex-col gap-0"
            >
              <div
                aria-label="Tabs"
                class="flex text-center flex-wrap -mb-px border-b border-gray-200 dark:border-gray-700"
                role="tablist"
              >
                <button
                  aria-controls=":r0:-tabpanel-0"
                  aria-selected="true"
                  class="flex items-center justify-center p-4 text-sm font-medium first:ml-0 disabled:cursor-not-allowed disabled:text-gray-400 disabled:dark:text-gray-500 focus:outline-none first:rounded-tl-md text-cyan-600 rounded-t-lg border-b-2 border-cyan-600 active dark:text-cyan-500 dark:border-cyan-500"
                  id=":r0:-tab-0"
                  role="tab"
                  style="z-index: 1;"
                  tabindex="-1"
                  type="button"
                >
                  group foobar
                </button>
              </div>
              <div
                class=""
              >
                <div
                  aria-labelledby=":r0:-tab-0"
                  class="py-0"
                  id=":r0:-tabpanel-0"
                  role="tabpanel"
                  tabindex="0"
                >
                  <div
                    class="relative"
                    data-testid="table-element"
                  >
                    <div
                      class="absolute bg-white dark:bg-gray-800 w-full h-full top-0 left-0 rounded-lg shadow -z-10"
                    />
                    <table
                      class="w-full text-left text-sm text-gray-500 dark:text-gray-400"
                    >
                      <thead
                        class="group/head text-xs uppercase text-gray-700 dark:text-gray-400"
                      >
                        <tr>
                          <th
                            class="group-first/head:first:rounded-tl-lg bg-gray-50 dark:bg-gray-700 px-6 py-3 group-first/head:last:rounded-tl-none group-first/head:last:rounded-tr-none"
                            colspan="6"
                            data-testid="unit foobar"
                          >
                            unit foobar
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="group/body"
                      >
                        <tr
                          class="group/row"
                          data-testid="user foobar"
                        >
                          <td
                            class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4 w-1/6 whitespace-normal !py-3 sm:whitespace-nowrap"
                          >
                            <div
                              class="flex items-center"
                            >
                              <div
                                class="ml-4 flex flex-col"
                              >
                                <a
                                  class="text-sm font-semibold hover:text-gray-600 active:text-blue-600 dark:hover:text-gray-500 dark:active:text-blue-600"
                                  href="/users/1"
                                >
                                  user foobar
                                </a>
                                <div
                                  class="text-xs md:hidden"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4 hidden w-1/6 whitespace-normal !py-3 md:table-cell lg:whitespace-nowrap"
                          />
                          <td
                            class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4 hidden w-1/6 whitespace-normal !py-3 md:table-cell lg:whitespace-nowrap"
                          />
                          <td
                            class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4 w-1/12 whitespace-nowrap !py-3 text-right md:text-center"
                          />
                          <td
                            class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4 hidden w-1/12 !py-3 text-center lg:table-cell"
                          >
                            <span
                              class="inline-flex rounded-full bg-sky-100 px-2 text-xs font-semibold leading-5 text-sky-600"
                            >
                              Offline
                            </span>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
      <footer
        class="flex flex-row justify-center"
      >
        <div
          class="mt-2 flex items-center space-x-1"
        >
          <svg
            class="h-6 w-16"
            height="50"
            version="1.2"
            viewBox="0 0 215 50"
            width="215"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              PERSCOM
            </title>
            <g
              id="background-logo"
            >
              <path
                d="m-45-102.3h366.9v292.6h-366.9z"
                id="Layer"
                style="fill: none;"
              />
            </g>
            <g
              id="text-logo-path-group"
            >
              <g
                id="Layer"
              >
                <g
                  id="text-logo-path-0"
                >
                  <path
                    d="m81.9 39.3l-7.2 1.4q4.3-18.7 5.8-29.1 4.5-1.3 6.7-1.3 2.2 0 3.5 0.6 1.2 0.6 2 1.7 1.5 1.9 1.5 5.5 0 2.2-0.9 4.4-0.8 2.2-2.2 3.8-1.5 1.7-3.3 2.7-1.9 1-3.8 1h-0.5q-0.8 3.3-1.6 9.3zm2.5-13.7q1.4-0.3 2.7-2.3 1.3-2.2 1.3-4.9 0-1.5-0.7-2.3-0.6-0.8-1.4-0.8-1.1 5.8-1.9 10.3zm19.8 13.1l-13.4 2 5.4-29.2 13.1-1.2q-0.3 1-0.5 1.8-0.2 0.9-0.4 1.7-0.2 0.9-0.3 1.5-1.2 0.1-3.3 0.1h-2.8l-1.5 7.4q2.2-0.1 3.7-0.1 1.5-0.1 2-0.1-0.2 1.2-0.3 2.3-0.2 1.1-0.4 2.2-0.4 0.1-1.3 0.1-1 0-2 0.1-1 0-2.2 0l-1.4 7.6 6-0.9zm7.2 0.3l-6.7 1.7q4.2-20.2 5.1-28.7 4.7-1.4 10.5-1.4 4.4 0 5.2 3 0.2 0.9 0.2 2.3 0 1.3-0.5 2.9-0.6 1.6-1.6 3.1-2 2.9-4.9 4.5 1.6 3.4 2.8 6.3 1 2.4 1.7 4.2l2.9-5.8q0.6 1.1 2.2 2 1.6 1 2.4 1 0.9 0 1.4-0.2 0.5-0.1 1-0.4 1-0.8 1-1.8 0-1-0.7-1.9-0.6-0.8-1.5-1.6-1-0.7-2.1-1.5-1.1-0.8-2-1.7-2.3-2-2.3-4.2 0-2.2 1.3-4.1 1.2-1.8 3.2-3.2 1.9-1.3 4.2-2.1 2.2-0.8 3.8-0.8 1.6 0 2.6 0.3 1 0.3 1.7 0.8 1.6 1.3 1.6 3.8 0 0.6-0.1 1.2-0.1 0.5-0.2 1.2l-6 1.8q0.2-0.8 0.2-1.7 0-0.9-0.4-1.5-0.5-0.5-1-0.5-0.5 0-1 0.2-0.5 0.3-0.9 0.7-1.2 0.9-1.2 2 0 1 0.6 1.7 0.6 0.7 1.4 1.2 0.8 0.6 1.8 1.2 0.9 0.6 1.8 1.5 1.9 2.3 1.9 6 0 4.5-3.3 7.5-3.1 2.8-7.9 2.8-3.4 0-6.1-3.3 0 0.2 0.1 0.3l-6.4 2.2q-3.2-10.9-3.3-12.4l-0.5 0.2q-1.3 7.4-2 11.2zm4.4-22.7l-1.4 6.6q1.7-0.2 4.1-2.1 2.6-1.9 2.6-3.1 0-1.2-2.3-1.4-0.6 0-1.2 0zm27.5 22.1q-1.6-2.5-1.6-6.6 0-4 1-7.8 1.1-3.8 3-6.8 2-3 4.6-4.8 2.7-1.9 5.9-1.9 2.1 0 3.5 1.6 1.4 1.6 1.4 4.2v0.5q0 0.3 0 0.6l-5.4 1.4q-0.5-1-1.2-1-0.6 0-1.4 0.8-0.7 0.7-1.4 1.9-0.7 1.3-1.3 2.8-0.7 1.5-1.2 3-1.1 3.4-1.1 5.4 0 2.1 1.3 2.1 1 0 2.1-1.2 1.1-1.3 1.4-2.7l4.8 1.9q-1.9 4.2-4.8 6.7-2.6 2.2-5.4 2.2-2.5 0-4.2-2.3zm14.7-6q0-4.6 2.2-9.9 2.1-5.3 5.6-8.9 3.7-3.7 7.8-3.7 3.4 0 5 2.7 1.2 2.2 1.2 6.7 0 2.5-0.7 5.4-0.7 3-1.9 5.5-1.2 2.5-2.8 4.5-1.6 1.9-3.3 3.3-3.6 2.8-6.9 2.8-2.8 0-4.4-2-1.8-2.2-1.8-6.4zm6.2-1.1q0 1.6 1.3 2.4 0.4 0.3 1 0.3 1.8 0 3.2-1.3 1.5-1.2 2.4-3.1 1-1.8 1.5-3.9 0.5-2.1 0.5-3.9 0-3.3-1.9-3.3-2.5 0-4.5 2.4-1.5 1.9-2.7 5.3-0.8 2.7-0.8 5.1zm18.2 8.7l-6.8 0.4q6.6-19.8 8.9-28.9 2.3 0 4.3-0.3 2-0.2 4.4-0.4-0.2 0.9-0.3 2-0.1 1.1-0.3 2.4-0.2 1.2-0.4 2.5-0.2 1.3-0.5 3.4-0.3 2.1-0.8 5.2h0.6q4.6-12.5 6.1-15.6 2.1-0.2 3.4-0.4 1.3-0.3 2.2-0.4 0.9-0.1 2-0.3-2 11.6-3.3 29.4l-7.2 1.2q0-1.4 0.4-4.6l0.5-4.8q0.4-3.7 0.8-5.7h-0.4q-0.5 1.7-1.2 3.6-0.8 1.9-1.5 3.9-0.7 2-1.4 3.9-0.7 1.9-1.2 3.6-1.5 0.1-2.5 0.1l-2.4 0.2 1.3-15.3h-0.6q-0.1 0.5-0.4 1.8-0.4 1.2-0.8 2.9-0.4 1.6-0.9 3.3-0.5 1.8-1 3.3-0.7 2.6-1 3.6z"
                    fill-rule="evenodd"
                    id="Layer"
                    style="fill: #646464; stroke: #ffffff; stroke-width: 0;"
                  />
                </g>
              </g>
            </g>
            <g
              id="shape-4ba72ef2-936a-49d3-93b3-1ec8e54791af"
            >
              <g
                id="Layer"
              >
                <g
                  id="path-1"
                >
                  <path
                    d="m51.2 14.9l-3 17.9c-0.8 4.9-3.6 9.2-7.6 11.5l-4.3 2.5-4.3-2.5c-4-2.3-6.9-6.6-7.7-11.5l-3-17.9 15-5.8zm-4.3-7.1l-0.9-0.9 1.3-0.2 0.5-1.1 0.6 1.1 1.3 0.2-0.9 0.9 0.2 1.3-1.2-0.6-1.1 0.6zm-12-4.3l-1.4-1.3 1.9-0.2 0.8-1.7 0.8 1.7 1.8 0.2-1.3 1.3 0.3 1.8-1.6-0.8-1.7 0.8zm-11.2 4.3l-0.9-0.9 1.3-0.2 0.5-1.1 0.6 1.1 1.3 0.2-0.9 0.9 0.2 1.3-1.2-0.6-1.1 0.6z"
                    id="Layer"
                    style="fill: #d9d9d9;"
                  />
                </g>
                <g
                  id="path-2"
                >
                  <path
                    d="m52.3 22.9l-0.3 1.4 15.2-4.7-4.3 7.7-12 3.7-0.2 1.4 11.4-3.6-4.3 7.7-9.4 2.9c-1.6 2.9-3.9 5.3-6.7 6.9l-5.4 3.2-5.5-3.2c-2.7-1.6-5-4-6.6-6.9l-9.4-2.9-4.3-7.7 11.4 3.6-0.3-1.4-12-3.7-4.3-7.7 15.2 4.7-0.2-1.4-15.9-4.9-4.2-7.7 19 5.9-0.5-2.8 17.6-6.8 17.5 6.8-0.4 2.8 19-5.9-4.3 7.7zm-2.8 10.1l3.2-18.9-16.4-6.3-16.4 6.3 3.2 18.9c0.9 5.3 4 9.9 8.3 12.4l4.9 2.8 4.9-2.8c4.3-2.5 7.4-7.1 8.3-12.4z"
                    fill-rule="evenodd"
                    id="Layer"
                    style="fill: #d9d9d9;"
                  />
                </g>
              </g>
            </g>
          </svg>
          <div
            class="text-xs dark:text-gray-500"
          >
            Widget Provided By
             
            <a
              class="underline"
              href="https://perscom.io"
              rel="noreferrer"
              target="_blank"
            >
              PERSCOM.io
            </a>
            .
             
            <span
              class="hidden sm:inline"
            >
              Copyright 
              2024
               Deschutes Design Group LLC.
            </span>
             
            <span
              class="hidden md:inline"
            >
              Version 
              1.0.0
              .
            </span>
          </div>
        </div>
      </footer>
    </body>
  </html>
</div>
`;
